sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityResultJsonWriter.java

```diff
@@ -15,28 +15,37 @@
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
-public class SensitivityValueJsonWriter implements SensitivityValueWriter, AutoCloseable {
+public class SensitivityResultJsonWriter implements SensitivityResultWriter, AutoCloseable {
 
     private final JsonGenerator jsonGenerator;
+    private final JsonGenerator jsonStatusGenerator;
 
-    public SensitivityValueJsonWriter(JsonGenerator jsonGenerator) {
+    public SensitivityResultJsonWriter(JsonGenerator jsonGenerator, JsonGenerator jsonStatusGenerator) {
         this.jsonGenerator = Objects.requireNonNull(jsonGenerator);
+        this.jsonStatusGenerator = Objects.requireNonNull(jsonStatusGenerator);
         try {
             jsonGenerator.writeStartArray();
+            jsonStatusGenerator.writeStartArray();
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
     @Override
-    public void write(int factorIndex, int contingencyIndex, double value, double functionReference) {
+    public void writeSensitivityValue(int factorIndex, int contingencyIndex, double value, double functionReference) {
         SensitivityValue.writeJson(jsonGenerator, factorIndex, contingencyIndex, value, functionReference);
     }
 
+    @Override
+    public void writeContingencyStatus(SensitivityAnalysisResult.SensitivityContingencyStatus status) {
+        SensitivityAnalysisResult.SensitivityContingencyStatus.writeJson(jsonStatusGenerator, status);
```

Comment:
Maybe a single file strategy can be easily implemented here by just storing status in that method and write all in the close method.

Reply:
@geofjamg It can. But it will output an heterogeneous format that will need a specific deserializer to read back (see SensitivityAnalysisToolTest.runJsonOutput). We will then have three possible formats  : SensitivityValue / ContingencyStatus separated, an hybrid SensitivityValue / ContingencyStatus with a custom deserializer to read back the output from the writer, and SensitivityAnalysisResult which handle status / contingency / factors. Is that okay for you ?

Reply:
Note : In fact object mapper has some convenient function to read heterogeneous arrays of data directly. So no need for a new custom deserializer (see last commit).